### PR TITLE
Move things from template

### PR DESCRIPTION
**Pull request by teressaeid**
In line with @crayment 's approach and @interstateone's comments on my template PR, I am moving some of the bare utilities function into `Astro` as the template is going to be barer.

I also have a `Navigation` version of template https://github.com/RobotsAndPencils/iOSProjectTemplate/compare/Navigation where I successfully reworked access modifiers, most of the classes and extensions were declared `public`.

EDIT: Access modifiers changed on template and kept as is here

### CODE REVIEW

File: Astro/Utils/Array+AstroGadgets.swift [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+            if var group = mutableResult[key] {
+                group.append(element)
+                mutableResult[key] = group
+            }
+            else {
+                mutableResult[key] = [element]
+            }
+            return mutableResult
+        }
+    }
```
Comment (crayment) [inline Astro/Utils/Array+AstroGadgets.swift]: I think there is now a [sufficiently capable version of this](https://developer.apple.com/documentation/swift/dictionary/2995342-init) in the standard library and so we don't need to add to Astro?

File: Astro/Utils/Bundle+AstroGadgets.swift [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+public extension Bundle {
+    
+    public var releaseVersionNumber: String? {
+        return self.infoDictionary?["CFBundleShortVersionString"] as? String
+    }
+    
+    public var buildVersionNumber: String? {
+        return self.infoDictionary?["CFBundleVersion"] as? String
+    }
```
Comment (crayment) [inline Astro/Utils/Bundle+AstroGadgets.swift]: I like these additions but I don't like it's not immediately obvious to me which version maps to what value in the plist. I think keeping the names more inline with the plist keys would make it quicker to understand what they are? I'm thinking `shortVersionString` and `bundleVersion`. What do you think?

Comment (teressaeid) [reply Astro/Utils/Bundle+AstroGadgets.swift]: Updated to match plist key name 👍

File: Astro/Utils/Array+AstroGadgets.swift (hunk 1 above)
Comment (crayment) [reply Astro/Utils/Array+AstroGadgets.swift]: Thanks! Can you update the REAMDE so it no longer mentions CaseIterable please?

**Comment by infiniteNIL** [comment]
Hey @crayment , can you finish this review?

File: Astro/Utils/Array+AstroGadgets.swift (hunk 1 above)
Comment (crayment) [reply Astro/Utils/Array+AstroGadgets.swift]: I still see the `group(by:)` function here? @teressaeid Do you have a commit that needs pushing?

File: Astro/Utils/Bundle+AstroGadgets.swift (hunk 2 above)
Comment (crayment) [reply Astro/Utils/Bundle+AstroGadgets.swift]: I still see the old names here too?

**Comment by teressaeid** [comment]
@crayment Not sure why the change are not on my even on my local. Updated with the latest updates to your comments. Pretty sure that commit got lost between the two computers 😡

**Pull request merged**